com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/GameObjectBuilder.java

```diff
@@ -212,6 +236,36 @@ private PrototypeDesc.Builder transformGo(IResource resource,
                 }
             }
             ComponentDesc.Builder compBuilder = ComponentDesc.newBuilder(cd);
+
+            // #3981
+            // migrate label scale from LabelDesc to ComponentDesc
+            if (isComponentOfType(cd, "label")) {
+                try {
+                    // find the label resource in one of two places:
+                    // * in the build directory if it is generated from an embedded label (see build step above)
+                    // * in the project
+                    IResource labelResource = project.getResource(cd.getComponent());
+                    if (!labelResource.exists()) {
+                        labelResource = labelResource.output();
+                    }
+                    FileReader reader = new FileReader(labelResource.getAbsPath());
+                    LabelDesc.Builder lb = LabelDesc.newBuilder();
+                    if (lb.hasScale()) {
+                        compBuilder.setScale(lb.getScale());
+                    }
+                }
+                catch(IOException e) {
+                    throw new CompileExceptionError(e);
+                }
+            }
+
+            // #3981
+            // if the component doesn't have a scale we set a default of v3(1.0)
+            // if we do not set a scale it will have a scale of v3(0.0) at runtime
+            if ((isComponentOfType(cd, "sprite") || isComponentOfType(cd, "label")) && !cd.hasScale()) {
+                compBuilder.setScale(defaultScale);
+            }
```

Comment:
Is it a good idea to limit this to just sprites and labels if we're going the route of adding the default scale in Bob?

Reply:
See above

Reply:
You're right!